src/ol/layer/Graticule.js

```diff
@@ -679,19 +704,40 @@ class Graticule extends VectorLayer {
     ];
 
     const centerLonLat = this.toLonLatTransform_(validCenter);
+    if (isNaN(centerLonLat[1])) {
+      centerLonLat[1] = Math.abs(this.maxLat_) >= Math.abs(this.minLat_) ?
+        this.maxLat_ : this.minLat_;
+    }
     let centerLon = clamp(centerLonLat[0], this.minLon_, this.maxLon_);
     let centerLat = clamp(centerLonLat[1], this.minLat_, this.maxLat_);
     const maxLines = this.maxLines_;
     let cnt, idx, lat, lon;
 
-    let validExtent = [
+    const validExtentP = [
```

Comment:
Again: what does the `P` mean?

Reply:
I tried to keep to the existing naming system where the P suffix was used for extents and coordinates in view projection units (so `validCenter` needs to be renamed `validCenterP`).  I think some comments would now be useful in `createGraticule_` and `updateProjectionInfo_` to explain the purpose of the various transforms and clamping.

Reply:
Ok, but then things like `trLonLatP` don't make sense, because the view projection won't be longitude/latitude except when the view projection is geographic coordinates.

Reply:
It's the top right of `[this.minLon_, this.minLat_, this.maxLon_, this.maxLat_]` transformed to view projection (which is not necessarily the top right of the transformed extent)

Reply:
Ok, then maybe `this.topRightP_`, `this.bottomLeftP` and so on? My point was that these values are not latitudes and longitudes.

Reply:
Neither is `this.maxLatP_` (it's the x coordinate of the right edge of the transformed extent). That is in the existing code. I didn't have any problem with that, `maxLat` refers to its source and `P` means it has been transformed, but perhaps it should be changed to `maxX` ?